src/util/eventbus.ts

```diff
@@ -0,0 +1,103 @@
+type Key = symbol | string;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type EventHandler = (...args: any[]) => void;
```

Comment:
There are some ways to build a type-safe event bus, for example:
https://basarat.gitbook.io/typescript/main-1/typed-event